### PR TITLE
test: Add Gitleaks workflow

### PR DESCRIPTION
**Pull request by chrkl**
## Description

Changes proposed in this pull request (what was done and why):

- Add Gitleaks workflow as a replacement for Prow job

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: .github/workflows/pull-gitleaks.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: pull-gitleaks
```
Comment (rakesh-garimella) [inline .github/workflows/pull-gitleaks.yml:1]:
```suggestion
name: gitleaks-detector
```
how about that ?

Comment (chrkl) [reply .github/workflows/pull-gitleaks.yml:1]: I would either keep it as it is to indicate that this entire file comes for the repository template or align with our naming and make it something like `PR Code checks / gitleaks`

Comment (rakesh-garimella) [reply .github/workflows/pull-gitleaks.yml:1]: `PR Code checks / gitleaks` sounds better

Comment (chrkl) [reply .github/workflows/pull-gitleaks.yml]: Done

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**